src/services/pull/usePullComponents.tsx

```diff
@@ -0,0 +1,207 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import {
+  FirstPullRequestSchema,
+  MissingBaseCommitSchema,
+  MissingBaseReportSchema,
+  MissingComparisonSchema,
+  MissingHeadCommitSchema,
+  MissingHeadReportSchema,
+} from 'services/comparison'
+import {
+  RepoNotFoundErrorSchema,
+  RepoOwnerNotActivatedErrorSchema,
+} from 'services/repo'
+import Api from 'shared/api'
+import A from 'ui/A'
+
+const query = `
+  query PullComponentsSelector(
+    $owner: String!
+    $repo: String!
+    $pullId: Int!
+    $filters: ComponentsFilters
+  ) {
+    owner(username: $owner) {
+      repository(name: $repo) {
+        __typename
+        ... on Repository {
+          pull(id: $pullId) {
+            compareWithBase {
+              ... on Comparison {
+                __typename
```

Comment:
Because the typename is under `... on Comparison` I don't believe it will show up for the other types, I would lift it up one level just in case.